packages/flutter/lib/src/widgets/framework.dart

```diff
@@ -3731,7 +3731,17 @@ class StatefulElement extends ComponentElement {
     assert(_state._debugLifecycleState == _StateLifecycle.created);
     try {
       _debugSetAllowIgnoredCallsToMarkNeedsBuild(true);
-      _state.initState();
+      final void _debugCheckForReturnedFuture = _state.initState();
```

Comment:
This will be illegal in Dart 2, I think. It might already fail our analyzer options?

Reply:
It doesn't :)

Reply:
I've updated the code to match the implementation of the setState return check (which does indeed use `as`).